packages/template-retail-react-app/app/components/badge-group/index.jsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import PropTypes from 'prop-types'
+import {useStyleConfig, Box, Badge} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {PRODUCT_TILE_BADGE_LABELS} from '@salesforce/retail-react-app/app/constants'
+
+const BadgeGroup = (props) => {
+    const {badgeLabels = PRODUCT_TILE_BADGE_LABELS, product} = props
+    const styles = useStyleConfig('BadgeGroup')
+
+    //TODO: check for duplicate labels
+    // If product is not provided/undefined then render the provided badge labels. This is useful if user already has the logic to get the badge labels outside the badge group
+    let filteredLabels = badgeLabels
+    if (product) {
```

Comment:
I think this might be why you asked the question about my pr about using the product object inside the component. 

I personally would recommend not having a product as a prop here. It makes the badge-group component less portable, e.g. you can now only use badge-groups with products. An example where you might want to use a badge group outside of the context of a product tile would be in the account section, maybe your users have an avatar, and that avatar could have something like "gold member" badge overlayed on it. 

In general, we strive to develop our components as tho we are creating a component library that is agnostic to being used in our demo site.